diff --git a/lib/ffi/struct_layout_builder.rb b/lib/ffi/struct_layout_builder.rb
index 918c790..ce80f67 100644
--- a/lib/ffi/struct_layout_builder.rb
+++ b/lib/ffi/struct_layout_builder.rb
@@ -87,7 +87,7 @@ def union?
     #   @return [0,1]
     #
     def packed=(packed)
-      if packed.is_a?(Fixnum)
+      if packed.is_a?(Integer)
         @alignment = packed
         @packed = packed
       else
